include/broker/zeek.hh

```diff
@@ -230,95 +230,43 @@ public:
   }
 
   bool valid() const {
-    if (as_vector().size() < 3)
-      return false;
-
-    auto vp = get_if<vector>(&(as_vector()[2]));
-
-    if (!vp)
+    auto&& outer = data_.to_list();
+    if (outer.size() < 3)
```

Comment:
This is kind of a general comment across the code in this file, but having these magic indexes that you're supposed to know what's in each of them feels bad to me. Is there no way for this to be a struct of some sort?

Reply:
Yeah, good call. It's a good opportunity to clean that up. I hope the last commit is more agreeable. 🙂

Reply:
That looks better to me 👍🏼